app/main/browser-window.js

```diff
@@ -71,15 +85,39 @@ export async function createBrowserWindow(userAgentClient, onload) {
     browser.webContents.send('user-agent-service-info', { port, host, version });
   });
 
+  // Only check and send the set default browser message if this is the first
+  // window opened, we're in a packaged build, on OSX (for now), and this isn't
+  // already the default browser.
+  const shouldCheckDefaultBrowser = browserWindows.length === 0 &&
+                                    process.env.NODE_ENV === 'production' &&
+                                    process.platform === 'darwin' &&
+                                    !protocols.isDefaultBrowser();
+
   // 'window-ready' is called if an error occurred loading the client, or once
   // the client has connected to the User Agent Service correctly.
   browser.once('window-ready', error => {
+    browser.IS_READY = true;
```

Comment:
Gonna let this one go for now:
1) Not sure how we can get around some sort of flag for BW's that are already ready -- store the once promise on the bw itself? `browser.once('window-ready', resolve); browser._ready = deferred` ?
2) I think this stuff will be changed when working on #1210, to handle BW lifetimes similar to Brave [here](https://github.com/brave/browser-laptop/blob/master/app/cmdLine.js)

Can change to whatever, actually prefer deferreds, but this code's going to be a bit transient for a few days